lib/core/utils.js

```diff
@@ -83,22 +83,24 @@ function retrieveEJSON() {
  * @param {(Topology|Server)} topologyOrServer
  */
 function maxWireVersion(topologyOrServer) {
-  if (topologyOrServer.ismaster) {
-    return topologyOrServer.ismaster.maxWireVersion;
-  }
+  if (topologyOrServer) {
+    if (topologyOrServer.ismaster) {
+      return topologyOrServer.ismaster.maxWireVersion;
+    }
 
-  if (typeof topologyOrServer.lastIsMaster === 'function') {
-    const lastIsMaster = topologyOrServer.lastIsMaster();
-    if (lastIsMaster) {
-      return lastIsMaster.maxWireVersion;
+    if (typeof topologyOrServer.lastIsMaster === 'function') {
+      const lastIsMaster = topologyOrServer.lastIsMaster();
+      if (lastIsMaster) {
+        return lastIsMaster.maxWireVersion;
+      }
     }
-  }
 
-  if (topologyOrServer.description) {
-    return topologyOrServer.description.maxWireVersion;
+    if (topologyOrServer.description) {
+      return topologyOrServer.description.maxWireVersion;
+    }
   }
 
-  return null;
+  return 0;
```

Comment:
Q: Why return null here?

Reply:
Mostly convenience. Version is numeric, and we usually test it with an expression like `maxWireVersion > n`.  Those tests would have to be `maxWireVersion && maxWireVersion > n` if it didn't return a numeric value.